modules/project/outputs.tf

```diff
@@ -40,7 +40,10 @@ output "default_service_accounts" {
 
 output "id" {
   description = "Project id."
-  value       = "${local.prefix}${var.name}"
+  value       = coalesce(
```

Comment:
Unfortunately I ran into this issue before vacation and I managed to fix it locally so I'm not able to reproduce right now. Somehow it looked to me like the explicit dependency on the project resource was not working in the output causing terraform to start configuring resources on project before it was created (some kind of race condition). I double checked with @LucaPrete (adding you just in case you remember something else that might be useful) and updating those output values solved my issue.